### PR TITLE
add parameter --max-epoch 

### PR DESCRIPTION
**Pull request by lfoppiano**
This PR adds the parameter --max-epoch which override the default value and disable Early stop. 

I've implemented only for sequence labelling

### CODE REVIEW

**Comment by kermitt2** [comment]
Hi Luca !

It seems in this PR that you consider that if `max-epoch` is defined, then the training must run without early stop?

However, it's not the case normally, we want to train with early stop until we reach a max epoch. I usually use it as max epoch threshold in early stop training too (to avoid training for too long). The training method for the different models implementation actually use early stop with max-epoch combined.

**Comment by lfoppiano** [comment]
Yes, I though that currently, the default values with earlyStop=True are good enough because it's going to stop anyway after 20-30 iterations. 
On the other hand, setting max_epoch, might indicate a clear intention on disabling the earlyStop. 
I can change it that it won't set earlyStop=False but then I will have to add an additional parameter to disable the earlyStop. 
What do you think?

**Comment by kermitt2** [comment]
> the default values with earlyStop=True are good enough because it's going to stop anyway after 20-30 iterations.

Well not necessarily, especially if patience is set higher.

>  On the other hand, setting max_epoch, might indicate a clear intention on disabling the earlyStop.

Personally I rely on max-epoch with early stop to avoid training too long. 

>  I can change it that it won't set earlyStop=False but then I will have to add an additional parameter to disable the earlyStop.

Personally I am not a fan of all these command line parameters (I prefer a config file or simply a dict somewhere, more readable), but yes that would be consistent with the training implementation.

**Comment by lfoppiano** [comment]
OK, then I will decouple early stop and max_epoch. 
To avoid misunderstanding I will add `--early-stop true` / `--early-stop false` to simply override the default (which depends on the model)

I agree that the number of parameters are getting a bit hard to follow.

**Comment by lfoppiano** [comment]
So far I've implemented number 1, because it seems less confusing, I can change it if requested
